applications: nrf_desktop: Drop MPSL flash sync timeout workaround (#20663)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * The ``dongle_small`` configuration for the nRF52833 DK.
    The configuration enables logs and mimics the dongle configuration used for small SoCs.

> * Removed implying a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL is used for synchronization between the flash memory driver and radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).

Reviewer: ```suggestion
* Removed implied, a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL is used for synchronization between the flash memory driver and radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).
```
Please check the meaning here as well.

>   The separate workqueue for connection TX notify processing used to be implied by the Bluetooth connectivity support (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).

Reviewer: ```suggestion
  The separate workqueue for connection TX notify processing is used as implied in the Bluetooth connectivity support (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).
```
Please check the meaning.

> * Removed implying a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL is used for synchronization between the flash memory driver and radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).
>   The separate workqueue for connection TX notify processing used to be implied by the Bluetooth connectivity support (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`).

Reviewer 2: Perhaps, this is a better way to phrase it

```suggestion
* Removed the imply statement from the nRF Desktop Bluetooth connectivity Kconfig option (:ref:`CONFIG_DESKTOP_BT <config_desktop_app_options>`) that enables a separate workqueue for connection TX notify processing (:kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ`) if MPSL is used for synchronization between the flash memory driver and radio (:kconfig:option:`CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL`).
```
